src/Orleans/Providers/ProviderTypeLoader.cs

```diff
@@ -87,6 +87,14 @@ private void ProcessAssemblyLocally(Assembly assembly)
 
         private static void ProcessNewAssembly(object sender, AssemblyLoadEventArgs args)
         {
+#if !NETSTANDARD
+            // If the assembly is loaded for reflection only avoid processing it.
+            if (args.LoadedAssembly.ReflectionOnly)
```

Comment:
When a provider dll is loaded as non-ReflectionOnly, it will call this again with that asm.

Eventually, we need to overhaul AssemblyLoaded and avoid ReflectionOnly